drivers/shared/executor/executor_linux.go

```diff
@@ -569,17 +551,44 @@ func (l *LibcontainerExecutor) handleExecWait(ch chan *waitResult, process *libc
 
 func configureCapabilities(cfg *lconfigs.Config, command *ExecCommand) error {
 	// TODO: allow better control of these
-	cfg.Capabilities = &lconfigs.Capabilities{
-		Bounding:    allCaps,
-		Permitted:   allCaps,
-		Inheritable: allCaps,
-		Ambient:     allCaps,
-		Effective:   allCaps,
+	// use capabilities list as prior to adopting libcontainer in 0.9
+	allCaps := supportedCaps()
+
+	// match capabilities used in Nomad 0.8
+	if command.User == "root" {
+		cfg.Capabilities = &lconfigs.Capabilities{
+			Bounding:    allCaps,
+			Permitted:   allCaps,
+			Effective:   allCaps,
+			Ambient:     nil,
+			Inheritable: nil,
+		}
+	} else {
+		cfg.Capabilities = &lconfigs.Capabilities{
+			Bounding: allCaps,
+		}
 	}
 
 	return nil
 }
 
+// supportedCaps returns a list of all supported capabilities in kernel
+func supportedCaps() []string {
+	allCaps := []string{}
+	last := capability.CAP_LAST_CAP
+	// workaround for RHEL6 which has no /proc/sys/kernel/cap_last_cap
```

Comment:
How did you discover this? We explicitly dropped support for RHEL6 in 0.9.0, so it's not necessary: https://github.com/hashicorp/nomad/blob/master/CHANGELOG.md#090-april-9-2019

Reply:
This function was moved from the initializer in https://github.com/hashicorp/nomad/blob/f9dbcd6e9e9a8ea3638b49e2e12f39a75d222c4c/drivers/shared/executor/executor_linux.go#L55-L65 - I didn't change it.

That being said, I suspect it's inspired by libcontainer cgroup handling (but with reverse logic) in https://github.com/hashicorp/nomad/blob/f9dbcd6e9e9a8ea3638b49e2e12f39a75d222c4c/vendor/github.com/moby/moby/daemon/caps/utils_unix.go#L15-L30 .